File: lswf.cpp
/* INPUT: 
 * 19
 * 
 * OUTPUT:
 * 1000101
 */
#include <iostream>
#include <fstream>
#define MAXG 1000

using namespace std;

int fibonacci(int* fib, int N) 
{
	fib[0] = 1; fib[1] = 1;
	int lst;
	
	for(int i = 2; i < N; i++) {
		fib[i] = fib[i-1] + fib[i-2];
		lst = i;
		if(fib[i] > N) break;
	}
	
	return lst;
}

int main()
{
	ifstream in("input.txt");
	ofstream out("output.txt");
	
	int N, i;
	in >> N;
	int caracts[MAXG], somma = 0, potSomma;
	int lastc;
	if(N > 4)
		lastc = fibonacci(caracts, N);
	else {
		caracts[0] = 1; caracts[1] = 1; caracts[2] = 2; caracts[3] = 3;
		if(N == 1) lastc = 1;
		else if(N == 2) lastc = 2;
		else if(N == 3) lastc = 3;
		else lastc = 4;
	}
	int* seq = new int[lastc];
	
<<<<<<< HEAD
	for(i = 0; i < lastc; i++) cout << caracts[i] << ' '; cout << endl;
=======
	for(i = 0; i < lastc; i++) cout << caracts[i] << endl; cout << endl;
>>>>>>> devs
	
	seq[0] = 1;
	for(i = lastc; i > 0; i--) {
		potSomma = somma + caracts[i];
		if(potSomma < N) {
			somma = potSomma;
			seq[i] = 1;
		} else seq[i] = 0;
	}
	for(i = 0; i < lastc; i++) out << seq[i];
	
	delete[] seq;
	in.close();
	out.close();
	return 0;
}

File: lswf.py
fib = [1, 1]
lst = 0


def fibonacci(N):
    global lst
    for i in range(2, N):
        fib.append(fib[i - 1] + fib[i - 2])
        lst = i
        if fib[i] > N:
            break


with open('input.txt', 'r') as fin:
    N = int(fin.readline())

fibonacci(N)

for i in fib:
   print(i, end=' ')

fib.reverse()
print()

somma = 0
seq = []
for i in range(0,len(fib)-1):
    potSomma = somma + fib[i]
    if potSomma < N:
        somma = potSomma
        seq.append(1)
    else:
        seq.append(0)

seq.append(1)
seq.reverse()

if N == 1 or N > 4:
    seq.pop()

with open('output.txt', 'w') as fout:
    for i in seq:
        fout.write(str(i))